pycbc/results/metadata.py

```diff
@@ -80,18 +80,31 @@ def save_png_with_metadata(fig, filename, fig_kwds, kwds):
 
     im.save(filename, "png", pnginfo=meta)
 
+def save_pdf_with_metadata(fig, filename, fig_kwds, kwds):
+    """Save a matplotlib figure to a PDF file with metadata.
+    """
+    # https://stackoverflow.com/a/17462125
+    from matplotlib.backends.backend_pdf import PdfPages
+
+    with PdfPages(filename) as pdfp:
+        fig.savefig(pdfp, format='pdf', **fig_kwds)
+        metadata = pdfp.infodict()
+        for key in kwds:
+            metadata[str(key)] = str(kwds[key])
+
```

Comment:
This is a bit more than improving the vt_ratio script, and I've not looked at this stuff before, but from the documentation this looks like it will do the right thing

Do you have an example output?

Reply:
PDF before/after:
https://ldas-jobs.ligo.caltech.edu/~tito.canton/pycbc/test_pr3529/vt_ratio_igwn.pdf
https://ldas-jobs.ligo.caltech.edu/~tito.canton/pycbc/test_pr3529/vt_ratio.pdf

PNG before/after:
https://ldas-jobs.ligo.caltech.edu/~tito.canton/pycbc/test_pr3529/vt_ratio_igwn.png
https://ldas-jobs.ligo.caltech.edu/~tito.canton/pycbc/test_pr3529/vt_ratio.png

I had no issues with these PDFs. The only caveat is that the metadata keywords I am inserting are not the "official" ones in the PDF standard, so I get an annoying warning like
```
lib/python2.7/site-packages/matplotlib/backends/backend_pdf.py:1579: UserWarning: Unknown infodict keyword: version
  warnings.warn('Unknown infodict keyword: %s' % k)
```
There seems to be a more modern way to add metadata to PDFs which matplotlib does not seem to support yet.

Reply:
Okay, thanks. I _may_ have had to rely on the pdf metadata being stored in plaintext in the raw bytes (apparently it is!), to check that it worked. These look good

Reply:
I added a couple lines that convert `title` to `Title`, the official PDF keyword for title. One warning less, and the title actually shows up in Preview and Firefox!